packages/subgraph/src/dao/dao_v1_0_0.ts

```diff
@@ -164,49 +163,52 @@ export function handleNativeTokenDeposited(event: NativeTokenDeposited): void {
 }
 
 export function handleGranted(event: Granted): void {
-  // ContractPermissionId
-  let daoId = event.address.toHexString();
-  let contractPermissionIdEntityId =
-    event.params.where.toHexString() +
-    '_' +
-    event.params.permissionId.toHexString();
-  let contractPermissionIdEntity = ContractPermissionId.load(
-    contractPermissionIdEntityId
-  );
-  if (!contractPermissionIdEntity) {
-    contractPermissionIdEntity = new ContractPermissionId(
-      contractPermissionIdEntityId
-    );
-    contractPermissionIdEntity.dao = daoId;
-    contractPermissionIdEntity.where = event.params.where;
-    contractPermissionIdEntity.permissionId = event.params.permissionId;
-    contractPermissionIdEntity.save();
-  }
+  const contractAddress = event.address.toHexString();
+  const where = event.params.where;
+  const contractPermissionId = event.params.permissionId;
+  const who = event.params.who;
+
+  const permissionEntityId = [
+    contractAddress,
+    where.toHexString(),
+    contractPermissionId.toHexString(),
+    who.toHexString()
+  ].join('_');
+
+  const daoAddress = contractAddress;
 
   // Permission
-  let permissionId =
-    contractPermissionIdEntityId + '_' + event.params.who.toHexString();
-  let permissionEntity = new Permission(permissionId);
-  permissionEntity.dao = daoId;
-  permissionEntity.contractPermissionId = contractPermissionIdEntityId;
-  permissionEntity.where = event.params.where;
-  permissionEntity.who = event.params.who;
-  permissionEntity.actor = event.params.here;
-  permissionEntity.condition = event.params.condition;
-  permissionEntity.save();
+  let permissionEntity = Permission.load(permissionEntityId);
+  if (!permissionEntity) {
+    permissionEntity = new Permission(permissionEntityId);
+    permissionEntity.where = event.params.where;
+    permissionEntity.permissionId = contractPermissionId;
```

Comment:
Why do we sometimes use the `event.params.` variables and sometimes the local `const ...` declared variables?

Reply:
The use of locally declared const variables is primarily to enhance readability and reduce repetitive access to event.params.*. By assigning event.params values to descriptive constants, the code becomes clearer, and there's no need to repeatedly reference event.params for the same values in subsequent lines.

Reply:
nice catch thanks, amended